File: baml_language/sdk_tests/crates/swift/type_shapes/customizable/roundtrip_tests/TestRouting.swift
// Roundtrip coverage for the cross-namespace routing-rules suite —
// port of python_pydantic2 `roundtrip_tests/test_routing.py`: root
// (`Baml`), `a`, `a.b`, `lorem`, and `ipsum` leaves.
//
// The `baml.http.Response` arms in `lorem` need an engine-minted handle;
// the handle round trip is covered by TestHandles.swift.
import XCTest
import Baml

final class TestRouting: XCTestCase {
    func test_routing_make_foo() throws {
        XCTAssertEqual(try Baml.make_foo(v: 3).v, 3)
    }

    func test_routing_round_trip_foo() throws {
        let f = Baml.Foo(v: 10)
        XCTAssertEqual(try Baml.round_trip_foo(f: f), f)
    }

    func test_routing_round_trip_thing_from_ab() throws {
        let t = Baml.a.b.Thing(v: 1)
        XCTAssertEqual(try Baml.a.b.round_trip_thing_from_ab(t: t), t)
    }

    func test_routing_round_trip_root_foo_from_ab() throws {
        let f = Baml.Foo(v: 2)
        XCTAssertEqual(try Baml.a.b.round_trip_root_foo_from_ab(f: f), f)
    }

    func test_routing_round_trip_deep_thing_from_a() throws {
        let t = Baml.a.b.Thing(v: 4)
        XCTAssertEqual(try Baml.a.round_trip_deep_thing_from_a(t: t), t)
    }

    func test_routing_round_trip_deep_thing_from_lorem() throws {
        let t = Baml.a.b.Thing(v: 5)
        XCTAssertEqual(try Baml.lorem.round_trip_deep_thing_from_lorem(t: t), t)
    }

    func test_routing_round_trip_resume() throws {
        let r = Baml.lorem.Resume(name: "ada", email: nil)
        XCTAssertEqual(try Baml.lorem.round_trip_resume(r: r), r)
    }

    func test_routing_round_trip_root_foo() throws {
        let f = Baml.Foo(v: 6)
        XCTAssertEqual(try Baml.lorem.round_trip_root_foo(f: f), f)
    }

    func test_routing_round_trip_lorem_resume_from_ipsum() throws {
        let r = Baml.lorem.Resume(name: "grace", email: "g@x.com")
        XCTAssertEqual(try Baml.ipsum.round_trip_lorem_resume_from_ipsum(r: r), r)
    }

    func test_routing_round_trip_resume_or_http_response() throws {
        let r = Baml.lorem.Resume(name: "lovelace", email: "a@x.com")
        XCTAssertEqual(try Baml.lorem.round_trip_resume_or_http_response(u: .t0(r)), .t0(r))
    }
}

File: baml_language/sdk_tests/crates/python_pydantic2/type_shapes/customizable/roundtrip_tests/test_routing.py
"""Roundtrip coverage for the cross-namespace routing-rules suite:
root (`baml_sdk`), `a`, `a.b`, `lorem`, and `ipsum` leaves.

The `baml.http.Response`-typed round trips in `lorem` are covered in
`test_streams.py` (they need an engine-minted handle and can't be built
host-side).
"""

import baml_sdk  # noqa: F401  — initializes the BAML runtime
from baml_sdk import Foo, make_foo, round_trip_foo
from baml_sdk.a import round_trip_deep_thing_from_a
from baml_sdk.a.b import Thing, round_trip_thing_from_ab, round_trip_root_foo_from_ab
from baml_sdk.lorem import (
    Resume,
    round_trip_resume,
    round_trip_root_foo,
    round_trip_deep_thing_from_lorem,
)
from baml_sdk.ipsum import round_trip_lorem_resume_from_ipsum


def test_routing_make_foo():
    assert make_foo(v=3).v == 3


def test_routing_round_trip_foo():
    f = Foo(v=10)
    assert round_trip_foo(f=f) == f


def test_routing_round_trip_thing_from_ab():
    t = Thing(v=1)
    assert round_trip_thing_from_ab(t=t) == t


def test_routing_round_trip_root_foo_from_ab():
    f = Foo(v=2)
    assert round_trip_root_foo_from_ab(f=f) == f


def test_routing_round_trip_deep_thing_from_a():
    t = Thing(v=4)
    assert round_trip_deep_thing_from_a(t=t) == t


def test_routing_round_trip_deep_thing_from_lorem():
    t = Thing(v=5)
    assert round_trip_deep_thing_from_lorem(t=t) == t


def test_routing_round_trip_resume():
    r = Resume(name="ada", email=None)
    assert round_trip_resume(r=r) == r


def test_routing_round_trip_root_foo():
    f = Foo(v=6)
    assert round_trip_root_foo(f=f) == f


def test_routing_round_trip_lorem_resume_from_ipsum():
    r = Resume(name="grace", email="g@x.com")
    assert round_trip_lorem_resume_from_ipsum(r=r) == r
